src/pages/Books/Books.tsx

```diff
@@ -1,51 +1,93 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import SearchRoundedIcon from '@mui/icons-material/SearchRounded';
 import clsx from 'clsx';
 
 import { Layout } from '../../components/Layout/Layout';
 import { BookCard } from '../../components/BookCard/BookCard';
 import { CurrentBookCard } from '../../components/CurrentBookCard/CurrentBookCard';
-import mockbooks from '../../data/mockbooks.json';
 import { Book } from '../../../types/types';
+import { BookAPI } from '../../api/bookApi';
+import { Pagination } from '../../components/Pagination/Pagination';
+import { Loading } from '../../components/Loading/Loading';
 import style from './Books.module.scss';
 
-const books: Book[] = mockbooks;
-
 export const Books: React.FC = () => {
   const [searchQuery, setSearchQuery] = useState('');
   const [isSearchOpen, setIsSearchOpen] = useState(false);
+  const [books, setBooks] = useState<Book[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [totalBooks, setTotalBooks] = useState(0);
 
-  const filteredBooks = books.filter(
-    (book) =>
-      book.title.toLowerCase().includes(searchQuery.toLowerCase()) ||
-      book.author.toLowerCase().includes(searchQuery.toLowerCase()),
-  );
+  const booksPerPage = 30;
+
+  const fetchBooks = async (page: number) => {
```

Comment:
If you would do

```suggestion
  const fetchBooks = async (page: number = 1) => {
```

you don't need to pass in the 1 in the Effect and then you can do

```
  useEffect(() => {
    const timeoutId = setTimeout(fetchBooks, 500);

    return () => clearTimeout(timeoutId);
  }, [searchQuery]);
```

If you like it or not, I will leave it completely up to you! Just wanna show you the possibilities 🤩